### PR TITLE
Replace modals with new Design System

### PR DESCRIPTION
**Pull request by martins87**
## Link to Issue
Closes: #4320 

### Added new Design System to following modals:

<details><summary>Feedback Modal</summary>

image

</details> 

<details><summary>Login Modal</summary>

image

</details> 

<details><summary>Delete Address Modal</summary>

image

</details> 

<details><summary>Delete Poll Modal</summary>

image

</details> 

<details><summary>Account Selector Modal</summary>

image

</details> 

<details><summary>TOS Modal</summary>

image

</details> 

<details><summary>Session Revalidation Modal</summary>

image

</details> 

<details><summary>Quill Preview Modal</summary>

image

</details> 

<details><summary>New Topic Modal</summary>

image

</details> 

<details><summary>Order Topics Modal</summary>

image

</details> 

<details><summary>Edit Topic Thresholds Modal</summary>

image

image

</details> 

<details><summary>Manage Contract Template Modal</summary>

image

</details> 

<details><summary>Canvas Verify Data Modal</summary>

image

</details> 

<details><summary>Edit Topic Modal</summary>

image

</details> 

<details><summary>Change Thread Topic Modal</summary>

image

</details> 

<details><summary>Edit Collaborators Modal</summary>

image

</details> 

<details><summary>Webhook Settings Modal</summary>

image

</details> 

<details><summary>Confirm Snapshot Vote Modal</summary>

image

</details> 

<details><summary>Update Proposal Status Modal</summary>

image

</details> 

<details><summary>Linked Thread Modal</summary>

image

</details> 

<details><summary>Poll Editor Modal</summary>

image

</details> 

<details><summary>Offchain Voting Modal</summary>

image

</details> 

<details><summary>New Snapshot Proposal Modal</summary>

image

</details> 

<details><summary>Template Action Modal</summary>

image

</details> 

## "How We Fixed It"

- Added reusable Header, Body and Footer components that since now should always be used to create new CWModal

## Test Plan
- click around the app, open modals and confirmation pop ups, make sure that they look good and works correctly

### CODE REVIEW

**Comment by zakhap** [comment]
Two things so far:
- Snapshot Creation Modal on Thread goes full screen oddly
- "Active State" Blue border is weird on some? also weird extra white space after border. 
Screen Shot 2023-08-10 at 1 51 43 PM
Screen Shot 2023-08-10 at 1 52 49 PM

**Review by masvelio — COMMENTED** [review]
While the modals look good for the user, the code is not perfect. The main problem is that every time the modal is used, its whole structure has to be build and styled over and over again. This means that for example `compact-modal-title` class is defined 21 times and each definition has 30 lines of code. What if design team decide to change the way the modal looks like? We have to change the markup and styling in 21 places. The same happens with `close-icon` which is repeated 23 times and other generic "blocks" that build the modal's interior. I really believe that we should come up with the clever solution, where you just use the modal and you dont care about styling its width and adding X icon because this should be available out of the box. This should be the main idea behind the design-system - to grab a component and use it, not copy over the code from other place where it looks similar cc @zakhap @jnaviask

**Comment by martins87** [comment]
> 

Thanks for the comment, Marcin. I'll get it reusable.

**Comment by martins87** [comment]
> 

Addressed.

A reusable 'CWModalHeader' component has been added to the `/modals` folder.
Label and Icon are optional, as pretty much all modals have a title but one or two do not and the header icon can appear as yellow warning or red danger.

The Acceptance Criteria of the ticket has been addressed with a reusable Modal Header.
@masvelio if this looks good I can extend this for the whole modal. Please let me know your thoughts.

Cc @zakhap @jnaviask @CowMuon

**Comment by martins87** [comment]
> Edit: If you can add a list of all instances of modals updated for review, that'd be 🔥 (just greaat)
> 
> Two things so far:
> 
> * Snapshot Creation Modal on Thread goes full screen oddly
> * "Active State" Blue border is weird on some? also weird extra white space after border.
> 
> Screen Shot 2023-08-10 at 1 51 43 PM Screen Shot 2023-08-10 at 1 52 49 PM

The max-width and max-height of the modals have been updated.

@zakhap I reproduced all modals here and didn't see this blue border. Can you please check again and let me know if it's going?

File: packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx [hunk 1]
```diff
@@ -11,22 +13,32 @@ type TOSModalProps = {
 };
 
 export const TOSModal = ({ onModalClose, onAccept }: TOSModalProps) => {
+  const chain = app.chain ? app.chain.meta : null;
+  const terms = app.chain ? chain.terms : null;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx:17]:
```suggestion
const terms = app.chain?.meta?.terms
```

File: packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx [hunk 2]
```diff
@@ -11,22 +13,32 @@ type TOSModalProps = {
...
-          By clicking accept you agree to the community's Terms of Service
+          By clicking accept you agree to the community's
+          <a href={terms} target="_blank">
+            Terms of Service
+          </a>
+          .
         </CWText>
-        <CWButton onClick={onAccept} label="Accept" />
+      </div>
+      <div className="compact-modal-footer">
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx:35]: you worked on header, but why not footer? it is duplicated dozens of times as well.

File: packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          <Warning className="warning-icon" weight="fill" size={24} />
+        )}
+        {icon === 'danger' && (
+          <WarningOctagon className="danger-icon" weight="fill" size={24} />
+        )}
+        <CWText className="title-text" type="h4">
+          {label}
+        </CWText>
+      </div>
+      <X className="close-icon" onClick={() => onModalClose()} size={24} />
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx:34]:
```suggestion
      <X className="close-icon" onClick={onModalClose} size={24} />
```

File: packages/commonwealth/client/styles/modals/CWModalHeader.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+@import '../shared';
+
+.compact-modal-title {
```
Comment (masvelio) [inline packages/commonwealth/client/styles/modals/CWModalHeader.scss:3]: top level class name should correspond with the component name

File: packages/commonwealth/client/styles/modals/linked_thread_modal.scss [hunk 5]
```diff
@@ -2,14 +2,16 @@
 
 .LinkedThreadModal {
   width: 504px;
+  max-height: 84vh;
+  max-width: 100vw;
```
Comment (masvelio) [inline packages/commonwealth/client/styles/modals/linked_thread_modal.scss:6]: again, these two lines are repeated like 20 times. And I am not sure why it is exactly 84vh? Should't we have fixed height of the modals?

**Review by masvelio — COMMENTED** [review]
1. Didn't we agree with @MuonShot and @jnaviask that we should write down some "tech spec" first on how we want to approach the modals revamp, then review it and start coding? 

2. Besides there are still things that are not handled properly, like repeated footers and sizes of the modals. So I am not sure if this PR is "ready" or half-ready / not-ready? 

 3. The ticket does not say it deliberately (maybe it should) but the purpose of the work is to create new design system component, in this case, modal. It means that in this ticket we should move cw_modal file to the new location and follow all new standards for the new design system.

**Comment by jnaviask** [comment]
@martins87 as discussed, please document your planned changes / technical approach on the ticket before continuing to update this PR

**Comment by martins87** [comment]
> @martins87 make it a draft for now

Done.

**Comment by martins87** [comment]
> @martins87 as discussed, please document your planned changes / technical approach on the ticket before continuing to update this PR

All 23 edited modals have a header section. Most of them have a header text; only some do not. This reusable header was implemented on [CWModalHeader](https://github.com/hicommonwealth/commonwealth/blob/daniel.4320.modals/packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx).

19 modals have three sections: header, body and footer, like the `BanUserModal` shown below. The footer on all of them have action buttons (e.g. `"Save"`, `"Cancel"`)

image

Some modals do not have a footer `div` but do have action buttons that are currently inside a `body` section. `ChangeThreadTopicModal` is an example, among 5 other modals:

image

These ones could have the action buttons moved to a footer section.

The solution I propose is a reusable modal that could follow all new standards for the new design system.
Body and Footer content could be props of this `CWModal` as `ReactNode` type, for example.

What do you think, @jnaviask, @MuonShot, @masvelio?

**Comment by masvelio** [comment]
Thanks for the research and your thoughts @martins87. My proposition of implementing this is as follows:

1. File structure:
   - Create new directory commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal
   ```
   - CWModalHeader
     -  CWModalHeader.tsx
     -  CWModalHeader.scss
   - CWModalBody
     - CWModalBody.tsx
     - CWModalBody.scss 
   - CWModalFooter
     - CWModalFooter.tsx
     - CWModalFooter.scss 
   - CWModal.tsx -> this is moved from current `cw_modal.tsx`
   - CWModal.scss
   - index.ts -> Exports CWModal, CWModalHeader, CWModalBody, CWModalFooter
   ```

2. CWModalHeader, CWModalBody, CWModalFooter should be composable and dummy components. Composable means that they can be used, but if they are not for some reason (eg there is custom design for specific modal), everything still works correctly. Dummy means that they should not hold the bussiness logic, they are containers that holds markup and styling. 

3. Usage

```
import { CWModal } from '../component_kit/new_designs/CWModal';

        <CWModal
          onClose={onClose}
          open={open}
          content={
            <PreviewModal
            // some props
            />
          }
        />
```

```
// PreviewModal.tsx

import { CWModalHeader, CWModalBody, CWModalFooter } from 'new_designs/CWModal'

export const PreviewModal = ({ all, neccessary, props }: PreviewModalProps) => {
  return (
    <div className="PreviewModal">
      <CWModalHeader label={label} icon={icon} onClose={onClose} />
      <CWModalBody>
        <div>Custom body.</div>
      </CWModalBody>
      <CWModalFooter>
        <Button>Submit</Button>
        <Button>Cancel</Button>
        // or any other custom elements
      </CWModalFooter>
    </div>
  );
};

```
Writing the modal, eg PreviewModal, I do not care about structure or styling because all of these is handled by CWModal, CWModalHeader, CWModalBody, CWModalFooter components. 
If there is no footer in the designs, I am simply not using CWModalFooter, but header and body still works fine.

4. If we have predefined (multiple) sizes of the modals, we should be able to pass it as a prop, not setting

File: packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx (hunk 2 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx]: Addressed. CWModalFooter added.

File: packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+type WarningIcon = 'warning' | 'danger';
+
+type CWModalHeaderProps = {
+  label?: string;
+  icon?: WarningIcon;
+  onModalClose: () => void;
+};
+
+export const CWModalHeader: FC<CWModalHeaderProps> = ({
```
[the remark this answers is not in the document]
Comment (martins87) [reply packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx]: Addressed.

File: packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx (hunk 3 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/modals/CWModalHeader.tsx]: Addressed.

File: packages/commonwealth/client/styles/modals/CWModalHeader.scss (hunk 4 above)
Comment (martins87) [reply packages/commonwealth/client/styles/modals/CWModalHeader.scss]: Addressed.

File: packages/commonwealth/client/styles/modals/linked_thread_modal.scss (hunk 5 above)
Comment (martins87) [reply packages/commonwealth/client/styles/modals/linked_thread_modal.scss]: It was defined on first spec on the ticket. Moved to parent CWModal component.

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModal.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+});
+
+const CWModal = (props: {
+  content: React.ReactNode;
+  isFullScreen?: boolean;
+  size?: Size;
+  onClose: () => void;
+  open: boolean;
+  className?: string;
+}) => {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModal.tsx:30]: change props => CWModalProps as a defined interface

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModalHeader/CWModalHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+const CWModalHeader: FC<CWModalHeaderProps> = ({
+  label = '',
+  subheader = '',
+  icon,
+  onModalClose,
+}) => {
+  return (
+    <>
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModalHeader/CWModalHeader.tsx:24]: I think <></> is not necessary here

**Review by masvelio — COMMENTED** [review]
Codewise it looks pretty good. I like the structure of the new components and I like that we can compose header, body and footer. 

Having said that, I think you need to go through every single modal and check how it looks like on various screen sizes. I can see right now that the content of modals does not stretch to the full width of parent container. 

image
image
In most cases there will be classes that set fixed width of the modal. This has to be removed, because since now we are gonna pass the size prop to the modal instead of handling the width of the modal in CSS.

Also, make sure that the modals do not have blue outline.
Other examples where modals are broken:
image
image
image
image

**Comment by masvelio** [comment]
@martins87 Please make sure that you focus on this PR and let's try to close it as soon as possible to avoid back and forth discussions and merge confclits.

File: packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx (hunk 1 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx]: [Addressed](https://github.com/hicommonwealth/commonwealth/pull/4757/files#diff-7de4c514233c73ba3ed5cd243c7d909d950a3e975ac803b0ade2ce6403076ca7R20).

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModal.tsx (hunk 7 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModal.tsx]: Addressed.

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModalHeader/CWModalHeader.tsx (hunk 8 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWModal/CWModalHeader/CWModalHeader.tsx]: Addressed.

**Comment by martins87** [comment]
> Codewise it looks pretty good. I like the structure of the new components and I like that we can compose header, body and footer.
> 
> Having said that, I think you need to go through every single modal and check how it looks like on various screen sizes. I can see right now that the content of modals does not stretch to the full width of parent container.
> 
> image image In most cases there will be classes that set fixed width of the modal. This has to be removed, because since now we are gonna pass the size prop to the modal instead of handling the width of the modal in CSS.
> 
> Also, make sure that the modals do not have blue outline. Other examples where modals are broken: image image image image

Addressed.

1. The content of the modals are now stretching to full width of parent container, with `size` being `small`, `medium`, `large` or `full width`.
2. Blue outline removed
3. The overflow on dropdowns have been fixed. Modals that have content that may overflow the modal (e.g. Dropdowns) have to pass the props `visibleOverflow` to CWModal component

image

image

File: packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx [hunk 9]
```diff
@@ -10,6 +10,7 @@ import { CWText } from 'views/components/component_kit/cw_text';
 import { isWindowMediumSmallInclusive } from 'views/components/component_kit/helpers';
 import { UserBlock } from 'views/components/user/user_block';
 import { LoginModal } from 'views/modals/login_modal';
+import { CWModal } from '../component_kit/new_designs/CWModal';
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx:13]: 'Modal' is defined but never used

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx [hunk 10]
```diff
@@ -14,6 +14,9 @@ import { Modal } from './cw_modal';
 import { CWTooltip } from './cw_popover/cw_tooltip';
 import { CWText } from './cw_text';
 import { getClasses } from './helpers';
+import { CWModal, CWModalBody, CWModalHeader } from './new_designs/CWModal';
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx:17]: image

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx (hunk 10 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx:15]: [Addressed](https://github.com/hicommonwealth/commonwealth/commit/3794fc26f0acfd7ed8bb37b4a0cdd62bdf6d0dc3).

File: packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx (hunk 9 above)
Comment (martins87) [reply packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx:12]: [Addressed](https://github.com/hicommonwealth/commonwealth/commit/3794fc26f0acfd7ed8bb37b4a0cdd62bdf6d0dc3).

**Review by masvelio — APPROVED** [review]
codewise looks good!

**Comment by martins87** [comment]
> codewise looks good!

Thanks for the review, @masvelio !

**Comment by martins87** [comment]
@jnaviask conflicts fixed.

**Comment by masvelio** [comment]
@martins87 I think failing tests are because of the changes in this PR. Also, do we have design/product sign-off? It would be great to get this in as it has been open for soo long

**Comment by martins87** [comment]
> @martins87 I think failing tests are because of the changes in this PR. Also, do we have design/product sign-off? It would be great to get this in as it has been open for soo long

@zakhap and @jessmart1213 can you take a look at the new Modals?

**Comment by masvelio** [comment]
@martins87 is this branch deployed anywhere? Where Jessica can see this? Also, the Test Plan section is not filled out.

**Comment by martins87** [comment]
> @martins87 is this branch deployed anywhere? Where Jessica can see this? Also, the Test Plan section is not filled out.

Test Plan section filled. Branch deployed to demo.

@jessmart1213 whenever you can, please test modals on Demo.

**Comment by masvelio** [comment]
@jnaviask I am gonna check every modal, make sure that it looks and works good. Then I will resolve the conflicts and we can merge it without the design signoff (as agreed with @jessmart1213 )

**Comment by masvelio** [comment]
I am around 10 modals in, around 15 left

**Comment by masvelio** [comment]
@jessmart1213 I went through all our modals in the app and made sure that they look good and works correctly. The list of modals with attached screenshots available in the description on top of this PR. As agreed, I am not gonna deploy it to the demo => design signoff should be done based on the above screenshots.

**Comment by jessmart1213** [comment]
### Design review

What a feat @masvelio ! So happy with these modals! Just a few adjustments! 

---

**New topic modal**
- Update to have a secondary button that says "Cancel"

**Order Topics Modal**
- Update to have a secondary button that says "Cancel"

**Ban User modal**
- Update Button copy to say "Ban address"
- Update to have a secondary button that says "Cancel"
- Add to content copy " This may take a few moments. Please click once and wait."

**Canvas Verify data modal**
- Please see image for reference. Because the primary action is to Edit, the "Delete topic" should be on the left and then "Cancel" and "Save Changes" should be on the right.
Screen Shot 2023-10-11 at 11 03 32 AM


**New snapshot proposal modal**
- Update button to be right aligned instead of on the left. 
- Update "Add voting choice" button to be a primary button (black not blue).
- Add a secondary button that says "Cancel" in modal footer.

**Comment by masvelio** [comment]
@jessmart1213 thanks for your feedback. Everything applied (and screenshots in the description updated as well).
@jnaviask can we prioritise this PR? I really want to get this out the door asap because of constant merge conflicts and how long this PR has been opened.

File: packages/commonwealth/client/styles/pages/login/login_mobile.scss [hunk 11]
```diff
@@ -114,7 +114,7 @@
   background-size: cover;
   display: flex;
   flex-direction: column;
-  height: 100%;
+  height: 100vh;
```
Comment (mzparacha) [inline packages/commonwealth/client/styles/pages/login/login_mobile.scss:117]: should we also set width to 100vw?

File: packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx [hunk 12]
```diff
@@ -111,7 +111,7 @@ export const LoginSelectorMenuLeft = () => {
       >
         <CWText type="caption">Connect a new address</CWText>
       </div>
-      <Modal
+      <CWModal
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/Header/LoginSelectorMenuLeft.tsx:114]: The `changeSelectedAddress` should be added to the dependency array. We can memoize `changeSelectedAddress`.

**Review by mzparacha — COMMENTED** [review]
Did the code review, and left some minor suggestions.

Will do the testing now.

**Comment by mzparacha** [comment]
Not sure if its by design but the close icon looks a bit dark on small screens for login modal

image

**Comment by mzparacha** [comment]
App crashes when clicking on the backdrop for delete address modal

https://github.com/hicommonwealth/commonwealth/assets/51641047/7e41df73-df00-4b27-986a-fdd3551c40e7

File: packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx [hunk 13]
```diff
@@ -1,33 +1,54 @@
...
-          className="close-icon"
+      <CWModalHeader
+        label="Terms of Service"
+        icon="warning"
+        onModalClose={onModalClose}
+      />
+      <CWModalBody>
+        <CWText className="body">
+          By clicking accept you agree to the community's
+          <a href={terms} target="_blank">
```
[the remark this answers is not in the document]
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx:32]: this is external link

**Comment by mzparacha** [comment]
The polls tab is empty on small screens (haven't checked this on master)

image
image

**Comment by mzparacha** [comment]
**Order topics modal**

Doesn't show a scroll bar if the topic's overflow (not sure if intentional). Just mentioning this since we added a custom scrollbar for the thread filter dropdown if the content overflows



image

image

**Comment by mzparacha** [comment]
**Poll editor modal**

Content overflows when more choices are added
image

**Comment by mzparacha** [comment]
This looks like a product/design question, just asking for visibility.

Shouldn't the header and footer of the modal content be fixed? Currently it scrolls with the body content.



image

image

**Comment by masvelio** [comment]
> Not sure if its by design but the close icon looks a bit dark on small screens for login modal

Login modal was not subject to any change in this scope. It got design sign-off

**Review by masvelio — COMMENTED** [review]
Applied first part of feedback

**Comment by masvelio** [comment]
@mzparacha ok went through all your comments and fixed everything in [this](https://github.com/hicommonwealth/commonwealth/pull/4757/commits/00d12099c9f9d973f09a52898a25cb62ef90532f) and [this](https://github.com/hicommonwealth/commonwealth/pull/4757/commits/b1008bd5d77c113a590e335612cd4fa9c7a334ec) commit. I resolved some comments that were obvious, rest I left so you can take a look and resolve if you want.

There are two things that I did not touch, and they are design-related. My take on this is that I would like to merge it because despite your points are valid, they are not blockers for this PR. So I am gonna reach out to the design team what is their opinion on this and if necessary, I will create some additional tickets. I am talking here about those two comments:
- [scroll not visible vs filter topic dropdown](https://github.com/hicommonwealth/commonwealth/pull/4757#issuecomment-1764620685)
-  [should footer be fixed vs scrolled](https://github.com/hicommonwealth/commonwealth/pull/4757#issuecomment-1764638327)

File: packages/commonwealth/client/styles/pages/login/login_mobile.scss (hunk 11 above)
Comment (mzparacha) [reply packages/commonwealth/client/styles/pages/login/login_mobile.scss:117]: Cool, I was thinking there might be a case where the height is fullscreen but the width doesn't cover full-screen width and shows the underlying screen on the left and right. But I couldn't reproduce it. Resolving

**Review by mzparacha — APPROVED** [review]
Awesome, Thank you for the changes @masvelio. GTG from my side.

**Pull request merged**